src/karmabot/bot.py

```diff
@@ -129,9 +130,10 @@ def karma_action(message, say):
     channel_id = message["channel"]
     karma_changes = KARMA_ACTION_PATTERN.findall(msg)
 
-    text = process_karma_changes(karma_giver, channel_id, karma_changes)
+    messages = process_karma_changes(karma_giver, channel_id, karma_changes)
 
-    say(text)
+    for message in messages:
+        say(message)
```

Comment:
inspector does have a point here, maybe use a different loop variable? it should be harmless in this code though

Reply:
just curious is every `say` an API call? if so what about doing a join on the messages (I guess by newline) and calling it once?

Reply:
whops, good point. did not even see that. 

yes, we could also format it and send everything out in one go

Reply:
nice